allennlp_models/generation/models/simple_seq2seq.py

```diff
@@ -66,21 +69,26 @@ class SimpleSeq2Seq(Model):
     """
 
     def __init__(
-        self,
-        vocab: Vocabulary,
-        source_embedder: TextFieldEmbedder,
-        encoder: Seq2SeqEncoder,
-        max_decoding_steps: int,
-        attention: Attention = None,
-        beam_size: int = None,
-        target_namespace: str = "tokens",
-        target_embedding_dim: int = None,
-        scheduled_sampling_ratio: float = 0.0,
-        use_bleu: bool = True,
-        bleu_ngram_weights: Iterable[float] = (0.25, 0.25, 0.25, 0.25),
+            self,
+            vocab: Vocabulary,
+            source_embedding_dim: int,
+            source_embedder: TextFieldEmbedder,
+            encoder: Seq2SeqEncoder,
+            max_decoding_steps: int,
+            attention: Attention = None,
+            beam_size: int = None,
+            decoder_layers: int = 2,
+            target_pretrain_file: str = None,
+            target_namespace: str = "tokens",
+            target_embedding_dim: int = None,
+            scheduled_sampling_ratio: float = 0.0,
+            use_bleu: bool = True,
+            bleu_ngram_weights: Iterable[float] = (0.25, 0.25, 0.25, 0.25),
     ) -> None:
         super().__init__(vocab)
+        self.source_embedding_dim = source_embedding_dim
```

Comment:
Why no underscore before `_source_embedding_dim`?

Reply:
Why did you add this parameter at all? Isn't it possible to get the source embedding dimension from the source embedder, without having to specify it? Also, if it's added here it needs to be added to the documentation.

Reply:
This parameter is useful when you have extra-features.
Suppose you have both word embedding(600 dim) and pos tags(600 dim). Then the embedding become (batch, length, 1200). But the encoder accept tensors with 600-dim. Then you may use a linear layer or directly add(just like style in BERT). This parameter will help.
But I have not finish all style "feature_merge" code in an elegant way.  **So I remove it.**
I will add some modules to finish it in the future.